packages/cli/src/fulltext/listGitProjectFIles.ts

```diff
@@ -1,42 +1,55 @@
-import { basename, join } from 'path';
-import { executeCommand } from '../lib/executeCommand';
-import { verbose, isFile } from '../utils';
+import { verbose } from '../utils';
+import { exec as execCb } from 'node:child_process';
+import { promisify } from 'util';
+
+const exec = promisify(execCb);
 
 // Run git ls-files and git status to get a list of all git-managed files. By doing it this way,
 // we automatically apply any .gitignore rules.
 export default async function listGitProjectFiles(directory: string): Promise<string[]> {
   const lsFiles = async (): Promise<string[]> => {
-    const gitFiles = (
-      await executeCommand('git ls-files', verbose(), verbose(), verbose(), [0], directory)
-    ).split('\n');
-    return gitFiles.filter(Boolean);
+    try {
+      const { stdout } = await exec('git ls-files', {
+        cwd: directory,
+        maxBuffer: 1024 ** 2 * 20, // 20 MB
```

Comment:
Is it possible to get the data in chunks so that we don't overflow the buffer and break the pipe? Maybe for such a large data set it doesn't matter anyway. But I'm curious.

Reply:
The existing [`executeCommand`](https://github.com/getappmap/appmap-js/blob/main/packages/cli/src/lib/executeCommand.ts) implementation does this, but it's somehow (unexpectedly) breaking the pipe. It's not immediately clear to me how this is happening, so I opted for this change.

```
Explain received context request: search
$ git ls-files
Command "git ls-files" killed by signal SIGPIPE, exited with code null
$ git status --porcelain
Error checking file Engine/Extras/Thi
[Error: ENOENT: no such file or directory, stat '/home/db/dev/EpicGames/UnrealEngine/Engine/Extras/Thi'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'stat',
  path: '/home/db/dev/EpicGames/UnrealEngine/Engine/Extras/Thi'
}
```